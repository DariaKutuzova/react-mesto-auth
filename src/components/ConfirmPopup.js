import PopupWithForm from "./PopupWithForm";


function ConfirmPopup({card, isOpen, onClose, onCardDelete}) {

    function handleSubmit(e) {

        e.preventDefault();

        onCardDelete(card);
    }


    return (
        <PopupWithForm
            name={'confirm'}
            title={'Вы уверены?'}
            isOpen={isOpen}
            onClose={onClose}
            onSubmit={handleSubmit}
        />
    );
}

export default ConfirmPopup;