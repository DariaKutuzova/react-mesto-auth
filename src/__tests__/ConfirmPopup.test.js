import React from 'react';
import renderer from 'react-test-renderer';
import ConfirmPopup from '../components/ConfirmPopup';

it('renders correctly', () => {
    const tree = renderer
        .create(<ConfirmPopup isOpen={true}
                              card={{
                                  "likes": [],
                                  "_id": "61bf8132842a2d00128a1ff1",
                                  "name": "gyer",
                                  "link": "https://www.imgonline.com.ua/examples/bee-on-daisy.jpg",
                                  "owner": {
                                      "name": "Мышka",
                                      "about": "НаруЖка00",
                                      "avatar": "https://images.wallpaperscraft.ru/image/single/shar_svechenie_siluet_141120_225x300.jpg",
                                      "_id": "bb397aec7c6ed8ff02690976",
                                      "cohort": "cohort-29"
                                  },
                                  "createdAt": "2021-12-19T19:00:02.903Z"
                              }}
                              onCardDelete={() => {
                              }}
                              onClose={() => {
                              }}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});