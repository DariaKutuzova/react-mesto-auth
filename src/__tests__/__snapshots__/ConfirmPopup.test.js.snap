// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="popup popup_type_confirm popup_opened"
  onClick={[Function]}
>
  <div
    className="popup__container popup__container_type_confirm"
  >
    <form
      action="#"
      className="popup__form popup__form_type_confirm"
      name="confirm"
      onSubmit={[Function]}
    >
      <h2
        className="popup__form-header"
      >
        Вы уверены?
      </h2>
      <button
        className="popup__button "
        type="submit"
      />
    </form>
    <button
      aria-label="Закрыть"
      className="popup__close popup__close_type_confirm"
      onClick={[Function]}
      type="button"
    />
  </div>
</div>
`;
